src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] tempArray;
+    private int size;
+
+    public ArrayList() {
+        tempArray = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void grow() {
+        int newLength = (int) (tempArray.length * 1.5);
+        T[] newTempArray = (T[]) new Object[newLength];
+        System.arraycopy(tempArray,0, newTempArray, 0, size);
+        tempArray = newTempArray;
+    }
+
+    private void indexCheck(int index) {
+        if (index >= 0 && index < size) {
+            return;
+        }
+        throw new ArrayListIndexOutOfBoundsException("Wrong index. Index should be: 0 <= index < "
+            + size);
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == tempArray.length) {
+            grow();
+        }
+        tempArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index. Should be: 0 <= index <= "
+                + size);
+        }
+        if (size == tempArray.length) {
+            grow();
+        }
+        System.arraycopy(tempArray, index, tempArray, (index + 1), (size - index));
+        tempArray[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list == null || list.isEmpty()) {
+            throw new NullPointerException("Provided List is empty.");
```

Comment:
```suggestion
            throw new RuntimeException("Provided List is empty.");
```

It better not throw NPE